tests/test_Summary.cpp

```diff
@@ -1178,6 +1195,18 @@ BOOST_AUTO_TEST_CASE(READ_WRITE_WELLDATA) {
             BOOST_CHECK_CLOSE( wellRatesCopy.get( "W_1" , rt::wat) , wellRates.get( "W_1" , rt::wat), 1e-16);
             BOOST_CHECK_CLOSE( wellRatesCopy.get( "W_2" , 101 , rt::wat) , wellRates.get( "W_2" , 101 , rt::wat), 1e-16);
 
+            const auto sm3_pr_day = unit::cubic(unit::meter) / day;
+
+            const auto& seg = wellRatesCopy.at("W_1").segments.at(1);
+            BOOST_CHECK_CLOSE(seg.rates.get(rt::wat),  123.45*sm3_pr_day, 1.0e-10);
```

Comment:
Any reason you do not continue with the full testing based on summary written to disk?

Reply:
At the time this was written, the layer that brought the values into the summary structure did not exist.  Additional unit tests are needed (and being created).